### PR TITLE
Add Rails 7.1.0 and Ruby 3.2.0 support

### LINKED ISSUE #1442: Support Rails 7.1
**Is your feature suggestion related to a problem? Please describe.**

With the recent release of Rails 7.1, it would be beneficial for this gem to be compatible with it to allow developers to take advantage of the latest features and improvements offered by Rails 7.1. 

**Describe the solution you'd like to build**

I suggest updating the gem to ensure compatibility with Rails 7.1. 
Currently they are as follows

lib/paper_trail/compatibility.rb
```ruby
ACTIVERECORD_LT = "< 7.1" # not enforced in gemspec
```

### PR DESCRIPTION
**Pull request by chrisyuska**
Thank you for your contribution!

Check the following boxes:

- [x] Wrote [good commit messages][1].
- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new 
  code introduces user-observable changes.
- [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

* Fixes #1442 

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by spaquet** [comment]
What is preventing this PR from making it to the main branch?

**Comment by danilogco** [comment]
@jaredbeck can you take a look on this, please?

File: spec/dummy_app/app/models/person.rb [hunk 1]
```diff
@@ -49,5 +49,10 @@ def load(value)
     end
   end
 
-  serialize :time_zone, TimeZoneSerializer.new
+  # Rails 7.1 deprecates positional argument for coder
+  if Rails.version >= "7.1"
```
Comment (StanBright) [inline spec/dummy_app/app/models/person.rb:53]: This check should be changed as we cannot compare version strings this way. At least, not correctly.

```ruby
"7.2.1" >= "7.10.1"
=> true

# Suggestion
Gem::Version.new(Rails.version) >= Gem::Version.new('7.1')
```

Comment (jermaine) [reply spec/dummy_app/app/models/person.rb:53]: Agreed! Could btw also be written like so:
Rails.gem_version >= Gem::Version.new("7.1")

**Review by jermaine — COMMENTED** [review]
All good to me 🚀

**Review by StanBright — COMMENTED** [review]
All good to me, too. Cheers.

**Comment by dfurber** [comment]
Seems good to go? @jaredbeck

**Comment by v-kumar** [comment]
Can we have a new release please? This is the only thing preventing us from upgrading to rails 7.1/7.1.1.

**Comment by bjorntrondsen** [comment]
@v-kumar you dont need to wait for a rubygems release to use this change

`gem 'paper_trail', git: 'git@github.com:chrisyuska/paper_trail.git', ref: '7309d7b9ed3256f9bbbafb0999bc140654b8a4c8'`

**Comment by spaquet** [comment]
@bjorntrondsen depends on your deployment process. some companies / project require to have a release in their Gemfile and not a reference to a tag in a branch ;-) 
It should not take too long for the team maintaining this project to release a new version (16) to offer support to Rails 7.1

**Comment by jaredbeck** [comment]
Merged offline. Thanks, @chrisyuska for your contribution :heart:

**Pull request closed without merging**